src/phoenix/trace/llama_index/callback.py

```diff
@@ -285,31 +299,47 @@ def _add_to_tracer(
     parent_child_id_stack: List[Tuple[Optional[SpanID], CBEventID]] = [
         (None, root_event_id) for root_event_id in trace_map["root"]
     ]
+    span_events: List[SpanEvent] = []
     while parent_child_id_stack:
         parent_span_id, event_id = parent_child_id_stack.pop()
         event_data = event_id_to_event_data[event_id]
         start_event = event_data["start_event"]
-        start_time_tz_naive = datetime.strptime(start_event.time, TIMESTAMP_FORMAT)
-        start_time_tz_aware = start_time_tz_naive.replace(tzinfo=_LOCAL_TZINFO)
-        end_event = event_data["end_event"]
-        end_time_tz_naive = datetime.strptime(end_event.time, TIMESTAMP_FORMAT)
-        end_time_tz_aware = end_time_tz_naive.replace(tzinfo=_LOCAL_TZINFO)
-        name = event_data["name"]
         event_type = event_data["event_type"]
+        if event_type is CBEventType.EXCEPTION:
+            if (
+                not start_event.payload
+                or (error := start_event.payload.get(EventPayload.EXCEPTION)) is None
+            ):
+                continue
+            span_events.append(
+                SpanException(
+                    message=str(error),
+                    timestamp=_timestamp_to_tz_aware_datetime(start_event.time),
+                    exception_type=type(error).__name__,
+                    exception_stacktrace=_get_stacktrace(error),
+                )
+            )
+            continue
+
+        start_time = _timestamp_to_tz_aware_datetime(start_event.time)
+        end_time = _get_end_time(event_data, span_events)
+        has_exception = len(span_events) > 0
+        name = event_data["name"]
         span_kind = _get_span_kind(event_type)
         span = tracer.create_span(
             name=name,
             span_kind=span_kind,
             trace_id=trace_id,
-            start_time=start_time_tz_aware,
-            end_time=end_time_tz_aware,
-            status_code=SpanStatusCode.OK,
+            start_time=start_time,
+            end_time=end_time,
+            status_code=SpanStatusCode.ERROR if has_exception else SpanStatusCode.OK,
             status_message="",
             parent_id=parent_span_id,
             attributes=event_data["attributes"],
-            events=None,
+            events=list(reversed(span_events)) or None,
```

Comment:
there's no requirement that they must be sorted, but if we do want it be sorted anyway, then it should be explicit.
```suggestion
            events=sorted(span_events, key=lambda event: event.timestamp),
```

Reply:
addressed in [4dd7df3](https://github.com/Arize-ai/phoenix/pull/1449/commits/4dd7df31606bcffbc8c7c42456646624351b44c2)